### PR TITLE
core: fix: follow scheduled point even if faster than margins

### LINKED ISSUE #8092: Scheduled point are not honored even if possible
### What happened?

Simulation tries to respect margins in priority even if scheduled points are not honored.

In practice, the function to compute how much time should be lost for the final_schedule is wrong. It should use the base simulation and add the calculated lost time, which may be smaller than the provisional time. 

### What did you expect to happen?

Scheduled points should be honored in priority.
If the lost time is smaller than the provided margins a warning will be sent back to the user.

### How can we reproduce it (as minimally and precisely as possible)?

1. Create a scenario using small infra
2. Import [this timetable](https://github.com/user-attachments/files/16263930/train_schedules.json)
3. See the final schedule is the same as the provisional (but a scheduled point is not honored)

### On which environments the bug occurs?

_No response_

### On which browser the bug occurs?

_No response_

### OSRD version (top right corner `Account` button > `Informations`)

769de63

### PR DESCRIPTION
**Pull request by eckter**
Fix https://github.com/OpenRailAssociation/osrd/issues/8092

We don't *quite* handle it perfectly yet: when we need to go faster than the margins, we ignore the time repartitions among different margin ranges (between the faulty pair of scheduled points).

### CODE REVIEW

File: core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt [hunk 1]
```diff
@@ -217,24 +222,51 @@ fun buildFinalEnvelope(
...
+            // we just set the time for the scheduled point without more details. It will be easier
+            // to handle it properly when we'll have migrated to standalone sim v3.
+            val maxEffortSectionTime =
+                getMaxEffortEnvelopeTimeAt(point.pathOffset) -
+                    getMaxEffortEnvelopeTimeAt(prevFixedPointOffset)
+            val earliestPossibleArrival = prevFixedPointDepartureTime + maxEffortSectionTime
+            var maxEffortExtraTime = point.arrival.seconds - earliestPossibleArrival
+            if (maxEffortExtraTime < 0.0) {
+                standaloneSimLogger.warn("impossible scheduled point")
+                // TODO: raise warning: scheduled point isn't possible
```
Comment (Khoyo) [inline core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt:253]: I'm not sure what this TODO is, considering the above line

File: core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt [hunk 2]
```diff
@@ -217,24 +222,51 @@ fun buildFinalEnvelope(
...
+                maxEffortExtraTime = 0.0
+            } else {
+                standaloneSimLogger.warn("scheduled point doesn't follow standard allowance")
+                // TODO: raise warning: scheduled point doesn't follow standard allowance
```
Comment (Khoyo) [inline core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt:257]: Ditto

**Review by Erashin — APPROVED** [review]

File: core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt (hunk 1 above)
Comment (eckter) [reply core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt:253]: The above line is just logging, we want that info to be forwarded to the user. It's related to https://github.com/OpenRailAssociation/osrd/issues/6925

**Review by Khoyo — APPROVED** [review]

**Review by shenriotpro — APPROVED** [review]
LGTM for tests/

**Pull request merged**